### ISSUE TITLE
Single-line change to Vertex() needed to enable compatibility with Godot 4.3+

### ISSUE THREAD

**Issue by Framebuffers** [GEG-fairbear8974/PSX-Style-Camera-Shader#2]
### Tested versions
Godot Engine v4.3.dev.mono.custom_build.ffe9ec6bf (2024-05-09 03:33:34 UTC) - https://godotengine.org
Vulkan 1.3.215 - Forward+ - Using Device #0: Intel - Intel(R) UHD Graphics 620

Works on any 4.3+ build.

### Issue
Similar to [Issue 88737](https://github.com/godotengine/godot/issues/88737). Related to changes on the shader API behind the scenes.

Upon loading the shader on 4.3+:

- Machine starts hanging up.
- Errors our in console with this message: `version_get_shader: Parameter "version" is null.`
- Lighting and other effects might get affected as well.

### How to reproduce

- Load the same test project provided under 4.3+ versions of Godot.
- Load the shader.
- Build. The following warning will appear on the console:

```
Line 48 (MAGIC_POSITION_WRITE):
You are attempting to assign the VERTEX position in model space to the vertex POSITION in clip space.
The definition of clip space changed in version 4.3, so if this code was written prior to 4.3, it will not continue to work.
Consider specifying the clip space z-component directly i.e. use `vec4(VERTEX.xy, 1.0, 1.0)`.
```

### Solution
- Deleting the .godot folder. Might be related to upgrading.
- Changing line 48 to `POSITION = vec4(VERTEX.xy, 1.0, 1.0);` as suggested by the warning.

![image](https://github.com/immaculate-lift-studio/PSX-Style-Camera-Shader/assets/26495556/4d5c9c3c-479d-407c-ad71-b45c97dc1845)

### Disclaimer
Yes, I built the engine from source. Yet the same issue documents some broken shaders on official dev builds of 4.3 (dev-6). It is, after all, new behaviour being introduced for everyone using shaders.

**Comment by immaculate-lift-studio** [comment]
Thanks for identifying the solution! Godot 4.3 introduces reverse z which adds this error and I haven't decided whether or not to fix it or simply remove the fog as the 4.3 update also adds distance-based fog back to the engine. I'm thinking about a couple solutions:

1. Fix the depth issue and release a 4.3 compatible version. 
2. Remove depth-based fog from the 4.3 version, instead relying on the built-in distance fog. The rest of the shader should work as advertised (dithering, color limiting, noise, etc). If the engine removes the fog or changes it in the future, it could break this shader.

Since the definition of clip space changed, I'm pretty sure I will need to maintain 2 versions, one legacy and one for the new definition. I'm open to both but I haven't taken the time to play around with 4.3 yet. Will make some time this month to explore it and reach a decision.

**Comment by immaculate-lift-studio** [comment]
Thank you for your help @Framebuffers. Pushed a 4.3-compatible version and added it to releases.
